src/os/FileStore.cc

```diff
@@ -641,10 +644,25 @@ bool parse_attrname(char **name)
 
 void FileStore::collect_metadata(map<string,string> *pm)
 {
+  char uuid_str[UUID_LEN+1];
+  blkid_cache cache = NULL;
+  blkid_dev dev = NULL;
+  uuid_d filestore_fsid = get_fsid();
```

Comment:
As much as we can trust any of the others in the code, as far as I can tell. :)

Basically, here's how it works:
1. if you use ceph-disk, the 'fsid' file that gets pushed down is the partition UUID for the underlying filesystem. That's our ideal scenario. Use ceph tools and you'll be able to manage things as expected.
2. if you don't use ceph-disk, you can create an OSD with a UUID that is completely independent from the partition UUID. Unfortunately, without popping out to a script and having more data about my original underlying device, I don't have a C/C++ mechanism to query the data I need. 

I think this is a reasonable compromise. Use ceph-disk/ceph-deploy, and everything will work as expected. Don't, and it's up to you to get the correct metadata in place.